lib/stripe_mock/request_handlers/events.rb

```diff
@@ -13,9 +13,36 @@ def retrieve_event(route, method_url, params, headers)
       end
 
       def list_events(route, method_url, params, headers)
-        Data.mock_list_object(events.values, params)
+        values = filter_by_created(events.values, params: params)
+        Data.mock_list_object(values, params)
       end
-      
+
+      private
+
+      def filter_by_created(event_list, params:)
+        if params[:created].nil?
+          return event_list
+        end
+
+        if params[:created].is_a?(Hash)
+          if params[:created][:gt]
+            event_list.select! { |event| event[:created] > params[:created][:gt].to_i }
+          end
+          if params[:created][:gte]
+            event_list.select! { |event| event[:created] >= params[:created][:gte].to_i }
+          end
+          if params[:created][:lt]
+            event_list.select! { |event| event[:created] < params[:created][:lt].to_i }
+          end
+          if params[:created][:lte]
+            event_list.select! { |event| event[:created] <= params[:created][:lte].to_i }
+          end
+        else
+          event_list.select! { |event| event[:created] == params[:created].to_i }
+        end
+        event_list
```

Comment:
You shouldn't use `select!` as this will modify the list and you haven't made a copy of it. Use `select` instead.

```suggestion
        if params[:created].is_a?(Hash)
          if params[:created][:gt]
            event_list.select { |event| event[:created] > params[:created][:gt].to_i }
          end
          if params[:created][:gte]
            event_list.select { |event| event[:created] >= params[:created][:gte].to_i }
          end
          if params[:created][:lt]
            event_list.select { |event| event[:created] < params[:created][:lt].to_i }
          end
          if params[:created][:lte]
            event_list.select { |event| event[:created] <= params[:created][:lte].to_i }
          end
        else
          event_list.select { |event| event[:created] == params[:created].to_i }
        end
```

Reply:
The list we're modifying came from `events.values`, which itself is a new copy—there's no need to `dup` it again.

Reply:
I went ahead and changed it though, so any future refactoring doesn't have to worry about it.